[free@home] Initial Contribution of free@home official local API

andrasU opened the pull request:
This is the initial contribution of the free@home binding, which is based on the official local API from ABB/Busch-Jäger

<!-- TITLE -->
ABB/Busch-Jäger free@home binding for OpenHAB

<!-- DESCRIPTION -->
This is a new binding for controlling free@home devices from ABB / Busch-Jaeger. I would like to provide this binding to the OpenHAB community to be able to bind the free@home devices through the official API.
This Binding makes possible to control the [free@home] devices through the free@home SysAP bridge.

The SysAP bridge is the only one device, which shall be added manually to the system. The following devices are supported

ABB / Busch-Jaeger System Access Point 2.0
free@home Switch Actuator Sensor 1/1, 2/1, 2/2 (wired and wireless)
free@home Dimming Actuator Sensor 1/1, 2/1 (wired and wireless)
free@home Blind Actuator Sensor 1/1, 2/1 (wired and wireless)
free@home Movement Detector Actuator
free@home Radiator Thermostat
free@home Switch Actuator 4-channel
free@home Switch Actuator 4-channel
free@home Dimming Actuator 4- and 6-channel
IP-touch panel (function: door opener, door ring sensor)
Hue devices (untested)

at this point I would like to say thank you for the User https://github.com/jannodeluxe for the support in the documentation

<!-- TESTING -->
[Link to the binary](https://github.com/andrasU/openhab-free-home-binding/blob/main/org.openhab.binding.freeathomesystem-3.1.0-SNAPSHOT.jar)

Skinah commented on bom/openhab-addons/pom.xml:
```diff
@@ -231,11 +231,13 @@
       <artifactId>org.openhab.binding.digiplex</artifactId>
       <version>${project.version}</version>
     </dependency>
+<!--
     <dependency>
       <groupId>org.openhab.addons.bundles</groupId>
       <artifactId>org.openhab.binding.digitalstrom</artifactId>
       <version>${project.version}</version>
     </dependency>
+-->
```
Why is this getting commented out?

andrasU replied:
On macOS Catalina 10.15.7 the full build is reporting error by these components. Shall I revert back?

andrasU replied:
changes are reverted

Skinah commented on bundles/org.openhab.binding.freeathomesystem/LICENSE:
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Any reason you do not want to use the same license that other bindings use?

andrasU replied:
removed

Skinah commented on bundles/org.openhab.binding.freeathomesystem/README.md:
```diff
@@ -0,0 +1,150 @@
+# OpenHAB 3 ABB/Busch-free@home Smart Home binding
```
1. It is meant to be openHAB, note the lower case o at start.
2. Please don't refer to a version number as this will out date when V4 is released.

```suggestion
# ABB/Busch-free@home Smart Home binding
```

Skinah commented on bundles/org.openhab.binding.freeathomesystem/README.md:
```diff
@@ -0,0 +1,150 @@
+![alt text](https://github.com/jannodeluxe/jannnnoooo/blob/main/abb_freeathome_2_0.png)
+# Description
+
+This openhab binding allows you to connect your free@home Smart Home system from ABB / Busch-Jaeger to OpenHAB and to control and observe most of the components.
```
Please change all occurences to use `openHAB`

andrasU replied:
all occurrences of openHAB corrected and committed to the branch

Skinah commented on bundles/org.openhab.binding.freeathomesystem/README.md:
```diff
@@ -0,0 +1,150 @@
+This openhab binding allows you to connect your free@home Smart Home system from ABB / Busch-Jaeger to OpenHAB and to control and observe most of the components.
+It requires a System Access Point with version 2.6.1 or higher.
+
+# Features
+
+* Control your ABB/Busch-free@home system with its Lights, Outlets, Blinds, etc. from OpenHAB via local API.
+
```
Remove all of this as it is already mentioned and is obvious. The supported devices goes into it in more detail.

Skinah commented on bundles/org.openhab.binding.freeathomesystem/README.md:
```diff
@@ -0,0 +1,150 @@
+
+* Control your ABB/Busch-free@home system with its Lights, Outlets, Blinds, etc. from OpenHAB via local API.
+
+# Supported devices
```
```suggestion
# Supported Devices
```

Skinah commented on bundles/org.openhab.binding.freeathomesystem/src/main/resources/OH-INF/thing/thing-types.xml:
```diff
@@ -0,0 +1,307 @@
+				<label>Shutter Actuator Up Trigger Channel</label>
+				<description>Shutter Actuator (UP) Trigger Channel for Free@Home Binding</description>
+			</channel>
+			<channel id="shutter_down_tigger_channel" typeId="shutter_trigger_channel_type">
```
```suggestion
			<channel id="shutter_down_trigger_channel" typeId="shutter_trigger_channel_type">
```
Is this a typo?

andrasU replied:
will be covered by the coding style changes

andrasU replied:
Correction is done with coding style fixes

Skinah commented on bundles/org.openhab.binding.freeathomesystem/src/main/resources/OH-INF/thing/thing-types.xml:
```diff
@@ -0,0 +1,307 @@
+	</thing-type>
+
+	<!--Thermostat temperature channel types -->
+	<channel-type id="thermostat_temp_channel_type">
```
All channels should use lowerCamelCase.
See https://www.openhab.org/docs/developer/guidelines.html#java-coding-style
Same comment for all channels that dont use the above.

andrasU replied:
Coding style fixes are done

Skinah commented on bundles/org.openhab.binding.freeathomesystem/src/main/resources/OH-INF/thing/thing-types.xml:
```diff
@@ -0,0 +1,307 @@
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!--Unknown Type -->
+	<thing-type id="unknown" listed="false">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge"/>
+		</supported-bridge-type-refs>
+		<label>Unknown</label>
+		<description>Unknown device for Free@Home Binding</description>
+	</thing-type>
```
What is the point of an unknown device? does it do anything?

andrasU replied:
The unknown device shall be used for further development to detect currently not implemented/unknown devices. This device detection is deactivated in the code. If it is activated generates a line in the discovery.
For standard user not visible.

Skinah replied:
So why does it need to be a real thing type? If I understand it is only for logging, you can just log there is an unknown device without there being a 'fake' non used `thing`? I probably dont understand what your trying to do here, hence why I ask. Just seems strange to create a fake thing that has no channels and does nothing.

andrasU replied:
During the development it was comfortable to visualise the devices directly in the openHAB web-frontend, instead read the long json files.
But you have completely right it is not necessary, it can be solved with logging the unknown devices.

andrasU replied:
Unknown device is replaced by logging of these unknown devices

andrasU commented:
Can I create a new PR, due to I made many changes since the last PR based on the current 3.2.0 version

> 2021. okt. 12. dátummal, 6:09 időpontban Matthew Skinner ***@***.***> írta:
> 
> 
> This PR is marked as a draft, when your ready for a review please change this and post that it is ready.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/openhab/openhab-addons/pull/10225#issuecomment-940638927>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ALIVDQO3L3HQK7XIE6YTJITUGOYHJANCNFSM4YFIPHBQ>.
>
